### PR TITLE
fix(one-app-runner): avoid shell usage

### PR DESCRIPTION
**Pull request by PixnBits**
## **Description**

Pass command arguments as arguments rather than allowing the shell to parse user input.

## **Motivation** 

>If the shell option is enabled, do not pass unsanitized user input to this function. Any input containing shell metacharacters may be used to trigger arbitrary command execution

https://nodejs.org/docs/latest-v18.x/api/child_process.html#child_processspawncommand-args-options

## **Test** **Conditions**

I used `$ npm pack` etc to test with a module and unit tests should be the same though inspecting the `args` instead of `command` now and moved to inline snapshots instead of putting them in another file.

## **Types of changes**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [x] Security update

## **Checklist**

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: packages/one-app-runner/src/startApp.js [hunk 1]
```diff
@@ -124,24 +162,75 @@ module.exports = async function startApp({
...
+    generateModuleMap(moduleMapUrl)
+  } ${
+    generateUseMocksFlag(parrotMiddlewareFile)
+  } ${
+    generateUseHostFlag()
+  }`;
+
+  try {
+    if (!offline) {
+      await dockerPull(appDockerImage);
```
Comment (PixnBits) [inline packages/one-app-runner/src/startApp.js:215]: can we use https://www.npmjs.com/package/dockerode instead? 🤔 

could also
```javascript
async function dockerCreateNetwork(name) {
  return new Promise((resolve, reject) => {
    const pullProcess = spawn('docker', ['network', 'create', name]);
    pullProcess.stdout.pipe(process.stdout);
    pullProcess.stderr.pipe(process.stderr);
    pullProcess.on('close', (code) => {
      if (code !== 0) {
        return reject(code);
      }
      return resolve(code);
    });
  });
}
```
and remove the dockerode dependency?

Comment (PixnBits) [reply packages/one-app-runner/src/startApp.js:215]: lines 20-78 of the diff are relevant context

**Review by Matthew-Mallimo — APPROVED** [review]

File: packages/one-app-runner/src/startApp.js [hunk 2]
```diff
@@ -124,24 +156,81 @@ module.exports = async function startApp({
...
+    devCDNPort,
+    devProxyServerPort,
+    metricsPort,
+    debugPort,
+  ];
+
+  const containerEnvVars = generateEnvironmentVariableArgs(envVars);
+
+  const mounts = new Map([
+    ...(modulesToServe || []).map((currentValue) => [currentValue, `/opt/module-workspace/${path.basename(currentValue)}`]),
```
Comment (PixnBits) [inline packages/one-app-runner/src/startApp.js:180]:
```suggestion
    ...(modulesToServe || []).map((moduleToServe) => [moduleToServe, `/opt/module-workspace/${path.basename(moduleToServe)}`]),
```

File: packages/one-app-runner/src/startApp.js (hunk 1 above)
Comment (PixnBits) [reply packages/one-app-runner/src/startApp.js]: I think we could remove dockerode usage, but that's probably best in a different PR

**Review by JAdshead — APPROVED** [review]

**Review by Matthew-Mallimo — APPROVED** [review]

**Pull request merged**